Model/BackendModel/GridCell.cs

```diff
@@ -135,17 +139,37 @@ private void WorldEntity_EntityMoved(object sender, EventArgs e)
 
         public void TakeSnapshot(int snapshotPointer)
         {
-            Snapshots[snapshotPointer] = JsonConvert.SerializeObject(this);
+            //TODO: There is a far more efficient method to do this... just let it slide for now
+
+            var prevSnapShot = Snapshots[(snapshotPointer + Snapshots.Length - 1) % Snapshots.Length];
+            var currentSnapshot = JsonConvert.SerializeObject(this);
+            if (prevSnapShot == currentSnapshot)
```

Comment:
Does simple == work here, or is it object equivalence?

Reply:
Oh nvm, snapshots are strings

Reply:
I got in a bad habit of using vars. I used to hate them when reading other's code.
But reshaper (or vs) tells you what type it is when hovering over, and slowly I became one _them_ 

I'll try to use proper variable types when it's not super obvious